cpp/include/cudf/aggregation.hpp

```diff
@@ -493,5 +495,74 @@ std::unique_ptr<Base> make_merge_sets_aggregation(null_equality nulls_equal = nu
 template <typename Base = aggregation>
 std::unique_ptr<Base> make_merge_m2_aggregation();
 
+/**
+ * @brief Factory to create a TDIGEST aggregation
+ *
+ * Produces a tdigest (https://arxiv.org/pdf/1902.04023.pdf) column from input values.
+ * The input aggregation values are expected to be fixed-width numeric types.
+ *
+ * The tdigest column produced is of the following structure:
```

Comment:
I'm still trying to wrap my head around the structure here. So an individual list can have multiple structs, one for each centroid? How many lists will there be?

Reply:
The # of lists (the # of rows) is == the # of groups.  Each group gets converted into an individual tdigest.  The length of each list is the "size" of each tdigest (the # of centroids in it).